File: ios/swift/uikit/youtube_tutorial/webapi/YoutubeWebApp/YoutubeWebApp/Models/Video.cs
using System.Text.Json.Serialization;

namespace YoutubeWebApp.Models
{
    public class Video
    {
        [JsonPropertyName("title")]
        public string Title { get; set; }

        [JsonPropertyName("number_of_views")]
        public int NumberOfViews { get; set; }

        [JsonPropertyName("thumbnail_image_name")]
        public string ThumbnailImageName { get; set; }

        [JsonPropertyName("channel")]
        public Channel Channel { get; set; }

        [JsonPropertyName("duration")]
        public int Duration { get; set; }
    }
}

File: ios/swift/uikit/youtube_tutorial/youtube_tutorial/youtube_tutorial/Models/Video.swift
//
//  Video.swift
//  youtube_tutorial
//
//  Created by Krzysztof Maraszkiewicz on 25/07/2020.
//  Copyright © 2020 Krzysztof Maraszkiewicz. All rights reserved.
//

import UIKit

class Video : NSObject {
    var thumbnail_image_name: String?
    var title: String?
    var number_of_views: NSNumber?
    var uploadDate: NSDate?
    var duration: NSNumber?
    
    var channel: Channel?
    
    init(dictionary: [String: Any]) {
        let channelDictionaty = dictionary["channel"] as! [String: Any]
        
        thumbnail_image_name = dictionary["thumbnail_image_name"] as? String ?? ""
        title = dictionary["title"] as? String ?? ""
        duration = dictionary["duration"] as? NSNumber ?? 0
        number_of_views = dictionary["number_of_views"] as? NSNumber ?? 0
        uploadDate = NSDate(timeIntervalSince1970: TimeInterval(exactly: 1000000)!)
        channel = Channel(name: channelDictionaty["name"] as! String,
                          profileImageName: channelDictionaty["profile_image_name"] as! String)
    }
}
